#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_metaflac_error_exit.dpatch by Joshua Kwan <joshk@triplehelix.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Exit with error if invalid option is passed. #329979

--- flac-1.1.2/src/metaflac/options.c~	2006-05-27 23:13:33.000000000 -0600
+++ flac-1.1.2/src/metaflac/options.c	2006-05-27 23:15:07.000000000 -0600
@@ -229,7 +229,7 @@
 		}
 	}
 
-	return !had_error;
+	return had_error;
 }
 
 void free_options(CommandLineOptions *options)
--- flac-1.1.2/src/metaflac/main.c~	2006-05-27 23:07:51.000000000 -0600
+++ flac-1.1.2/src/metaflac/main.c	2006-05-27 23:14:47.000000000 -0600
@@ -32,7 +32,7 @@
 	setlocale(LC_ALL, "");
 	init_options(&options);
 
-	if(parse_options(argc, argv, &options))
+	if ((ret = parse_options(argc, argv, &options)) == 0)
 		ret = !do_operations(&options);
 
 	free_options(&options);
